application/src/main/java/org/togetherjava/tjbot/features/help/HelpSystemHelper.java

```diff
@@ -131,6 +143,90 @@ private RestAction<Message> sendExplanationMessage(GuildMessageChannel threadCha
+     */
+    RestAction<Message> constructChatGptAttempt(ThreadChannel threadChannel,
+            String questionFirstMessage) {
+
+        Optional<String> questionOptional =
+                prepareChatGptQuestion(threadChannel, questionFirstMessage);
+        Optional<String> chatGPTAnswer;
+
+        try {
+            String question = questionOptional.orElseThrow();
+            logger.debug("The final question sent to chatGPT: {}", question);
+            chatGPTAnswer = chatGptService.ask(question);
+
+            String aiResponse = chatGPTAnswer.orElseThrow();
+            String lowercaseAiResponse = aiResponse.toLowerCase(Locale.US);
+            if (lowercaseAiResponse.contains("as an ai language model")) {
+                logger.debug("Response from ChatGPT, lacks context or unable to respond?: {}",
+                        aiResponse);
+                return useChatGptFallbackMessage(threadChannel);
+            }
+
+            UnaryOperator<String> response =
+                    """
+                            Here is an AI assisted attempt to answer your question 🤖. Maybe it helps! \
+                            In any case, a human is on the way 👍. To continue talking to the AI, you can use \
+                            %s.
+                            """::formatted;
+
+            return mentionGuildSlashCommand(threadChannel.getGuild(), ChatGptCommand.COMMAND_NAME)
+                .map(response)
+                .flatMap(threadChannel::sendMessage)
+                .flatMap(embed -> threadChannel
+                    .sendMessageEmbeds(HelpSystemHelper.embedWith(aiResponse)));
+        } catch (NoSuchElementException e) {
+            logger.warn("Something went wrong while trying to communicate with the ChatGpt API: {}",
+                    e.getMessage());
+            return useChatGptFallbackMessage(threadChannel);
+        }
+    }
+
+    private Optional<String> prepareChatGptQuestion(ThreadChannel threadChannel,
+            String questionFirstMessage) {
+        String questionTitle = threadChannel.getName();
+        StringBuilder questionBuilder = new StringBuilder(MAX_QUESTION_LENGTH);
+
+        if (questionFirstMessage.length() < MIN_QUESTION_LENGTH
+                && questionTitle.length() < MIN_QUESTION_LENGTH) {
+            return Optional.empty();
+        }
+
+        if (questionTitle.length() + questionFirstMessage.length() > MAX_QUESTION_LENGTH) {
+            // - 1 for the space that will be added afterward.
+            questionFirstMessage = questionFirstMessage.substring(0,
+                    (MAX_QUESTION_LENGTH - 1) - questionTitle.length());
+        }
+
+        questionBuilder.append(questionTitle).append(" ").append(questionFirstMessage);
+        for (ForumTag tag : threadChannel.getAppliedTags()) {
+            if (questionBuilder.length() > MAX_QUESTION_LENGTH) {
+                break;
+            }
+            questionBuilder.insert(0, String.format("%s ", tag.getName()));
```

Comment:
inserts are really painful, performance-wise. id suggest u create another stringbuilder to first construct the string and then have a single `insert` into ur target, instead of multiple.